pure-ftpd-1.0.36: Free (BSD), secure, production-quality and standard-conformant FTP server

diff --git a/pkgs/servers/ftp/pure-ftpd/default.nix b/pkgs/servers/ftp/pure-ftpd/default.nix
new file mode 100644
index 0000000..24b774e
--- /dev/null
+++ b/pkgs/servers/ftp/pure-ftpd/default.nix
@@ -0,0 +1,37 @@
+{ stdenv, fetchurl, openssl
+, mysqlSupport ? false, mysql ? null
+}:
+
+assert mysqlSupport -> mysql != null;
+
+let
+  pkg = "pure-ftpd";
+  ver = "1.0.36";
+in stdenv.mkDerivation rec {
+  name = "${pkg}-${ver}";
+
+  src = fetchurl {
+    url = "http://download.pureftpd.org/pub/pure-ftpd/releases/${pkg}-${ver}.tar.gz";
+    sha256 = "0c51nf7gxzs1sajfj9680f7gjrc9555wgqzkm5m0fj6lm6qn7ywh";
+  };
+
+  buildInputs = [ openssl ]
+    ++ stdenv.lib.optional mysqlSupport mysql;
+
+  configureFlags = ''
+    --with-ftpwho --with-quotas
+    --with-ratios --with-puredb --with-altlog --with-throttling
+    --with-privsep --with-tls
+    ${if mysqlSupport then "--with-mysql=${mysql}" else ""}
+  '';
+
+  meta = {
+    description = "Free (BSD), secure, production-quality and standard-conformant FTP server";
+    homepage = http://www.pureftpd.org/project/pure-ftpd;
+    license = stdenv.lib.licenses.mit.shortName;
+    platforms = stdenv.lib.platforms.all;
+    maintainers = with stdenv.lib.maintainers; [ tzeman ];
+  };
+}
+
+# vim: et ts=2 sw=2 
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 2ad8d14..2734234 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9876,6 +9876,11 @@ let
 
   pumpio = callPackage ../servers/web-apps/pump.io { };
 
+  pure-ftpd = callPackage ../servers/ftp/pure-ftpd {
+    mysqlSupport = true;
+    mysql = mysql55;
+  };
+
   pyIRCt = callPackage ../servers/xmpp/pyIRCt {};
 
   pyMAILt = callPackage ../servers/xmpp/pyMAILt {};
